src/keystore/persistence/EncryptedPersistence.ts

```diff
@@ -0,0 +1,82 @@
+import { Persistence } from './interface'
+import eccrypto, { Ecies } from 'eccrypto'
+
+const assertEciesLengths = (ecies: Ecies): void => {
+  if (ecies.iv.length !== 16) {
+    throw new Error('Invalid iv length')
+  }
+  if (ecies.ephemPublicKey.length !== 65) {
+    throw new Error('Invalid ephemPublicKey length')
+  }
+  if (ecies.ciphertext.length < 1) {
```

Comment:
AES produces blocks of 128bits -- I would expect `ciphertext.length % 16 ==0`

Reply:
I can add a check for that